### PR TITLE
kubeadm: update notes and example config in v1beta1 godocs

### LINKED ISSUE #911: Create a v1beta1 version for kubeadm config API types
Follow up from https://github.com/kubernetes/kubeadm/issues/750#issuecomment-396679842.

> For beta, we need to
> - update the KEP to reflect the current state
> - improve the structure of our API types further (gdocs proposal)
> - decide whether to embed other components' ComponentConfig in our types or not

There is way more needed here wrt details, but I don't have time to write up that all now, it will go into the KEP and be discussed during kubeadm Office Hours in the v1.12 cycle.

**Edit:** The KEP gdoc proposal for this is [here](https://docs.google.com/document/d/1Sk2VW7IKaLjrjf_4a1NEU0l7X0ysZoXq9lVBvH-ruYY/edit#heading=h.vd99op8cffst)

### LINKED ISSUE #1228: v1beta1: update and verify the godocs page
we need to update and verify the godocs page once all our refactoring is in place.

https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go

xref https://github.com/kubernetes/kubeadm/issues/911

### PR DESCRIPTION
**Pull request by neolit123**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
this PR updates the godocs page for kubadm changes in v1beta1.
PTAL and let's fix any typos and missing items.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

NOTE: this is the last PR for v1beta1 and it closes the following issues.

Fixes kubernetes/kubeadm#911
Fixes kubernetes/kubeadm#1228

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/assign @fabriziopandini @rosti 
@kubernetes/sig-cluster-lifecycle-pr-reviews 
/priority important-soon

### CODE REVIEW

**Review by timothysc — APPROVED** [review]
/approve

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go [hunk 1]
```diff
@@ -19,14 +19,24 @@ limitations under the License.
...
+// - "apiServerEndpoint" in InitConfiguration was renamed to "localAPIServerEndpoint" for better clarity of what the field
+// represents.
+// - Common fields in ClusterConfiguration such as "*extraArgs" and "*extraVolumes" for control plane components are now moved
+// under component structs - i.e. "apiServer", "controllerManager", "scheduler".
+// - "auditPolicy" was removed from ClusterConfiguration. Please use "extraArgs" in "apiServer" to configure this feature instead.
+// - "unifiedControlPlaneImage" in ClusterConfiguration was changed to a boolean field called "useHyperKubeImage".
+// - ClusterConfiguration now has a "dns" field which can be used to select and configure the cluster DNS addon.
+// - Both "localEtcd" and "dns" configurations now support custom image repositories.
+// - the "controlPlane*" related fields in JoinConfiguration were refactored into a sub structure.
+// - "clusterName" was removed from JoinConfiguration and the name is now fetched from the existing cluster.
```
Comment (rosti) [inline cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go]: We should probably mention `featureGates` too here.

**Review by rosti — COMMENTED** [review]
Thanks @neolit123 ! Looks good!

Should we keep the referenced issues for tracking changes to the website repo?

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go]: i did not mention them as we already have release notes on them and the "featureGates" element has not changed as a container for them.
but we can definitely do that, i guess.

**Comment by neolit123** [comment]
xref: https://github.com/kubernetes/website/issues/10937

**Review by timothysc — APPROVED** [review]
/lgtm

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go [hunk 2]
```diff
@@ -19,14 +19,24 @@ limitations under the License.
...
 
-// Package v1beta1 defines the v1beta1 version of the kubeadm config file format, that is a big step
-// forward the objective of graduate kubeadm config to beta.
-//
-// //TODO add notes about big changes introduced by this release
+// Package v1beta1 defines the v1beta1 version of the kubeadm config file format.
+// This version graduates the kubeadm config to BETA and is a big step towards GA.
+//
+// A list of changes since v1alpha3:
+// - "apiServerEndpoint" in InitConfiguration was renamed to "localAPIServerEndpoint" for better clarity of what the field
```
Comment (seh) [inline cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go]: Should that be "localAPIEndpoint?"

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go [hunk 3]
```diff
@@ -19,14 +19,24 @@ limitations under the License.
 // +k8s:deepcopy-gen=package
 // +k8s:conversion-gen=k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm
 
-// Package v1beta1 defines the v1beta1 version of the kubeadm config file format, that is a big step
-// forward the objective of graduate kubeadm config to beta.
-//
-// //TODO add notes about big changes introduced by this release
+// Package v1beta1 defines the v1beta1 version of the kubeadm config file format.
```
Comment (seh) [inline cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go]: s/config/configuration/

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go [hunk 4]
```diff
@@ -19,14 +19,24 @@ limitations under the License.
...
+// This version graduates the kubeadm config to BETA and is a big step towards GA.
```
Comment (seh) [inline cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go]: s/config/configuration format/

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go [hunk 5]
```diff
@@ -19,14 +19,24 @@ limitations under the License.
...
+// A list of changes since v1alpha3:
+// - "apiServerEndpoint" in InitConfiguration was renamed to "localAPIServerEndpoint" for better clarity of what the field
+// represents.
+// - Common fields in ClusterConfiguration such as "*extraArgs" and "*extraVolumes" for control plane components are now moved
+// under component structs - i.e. "apiServer", "controllerManager", "scheduler".
+// - "auditPolicy" was removed from ClusterConfiguration. Please use "extraArgs" in "apiServer" to configure this feature instead.
+// - "unifiedControlPlaneImage" in ClusterConfiguration was changed to a boolean field called "useHyperKubeImage".
+// - ClusterConfiguration now has a "dns" field which can be used to select and configure the cluster DNS addon.
+// - Both "localEtcd" and "dns" configurations now support custom image repositories.
+// - the "controlPlane*" related fields in JoinConfiguration were refactored into a sub structure.
```
Comment (seh) [inline cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go]: s/the/The/
s/" related/"-related/
s/sub structure/sub-structure/

**Comment by neolit123** [comment]
@seh a bit delayed, but thanks. :)
i wish more eyes looked at these. we can hopefully batch with other changes in the file.

**Comment by neolit123** [comment]
no problem, ideally if you manage to edit the whole page please create a tracking issue in k/kubeadm or even better - send a PR with the edits!
thanks.

**Comment by neolit123** [comment]
i just found another problem in the godoc:
under local etcd there is an `image: "k8s.gcr.io/etcd-amd64:3.2.18"` which is wrong should be separate repo / tag:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go#L160

**Comment by seh** [comment]
It looks like #71219 picks up my suggestions.

**Pull request merged**